itest/data/srv-configs/backend.main.yaml

```diff
@@ -39,6 +39,22 @@ cached_endpoints:
         id_identifier: 'id'
         pattern: "(^/not_authorized\\?ids=)((?:\\d|%2C)+)(.*$)"
         ttl: 30
+    # POST endpoints cached.
+    post_no_id:
```

Comment:
I'd add a new "method" field which defaults to "GET". That way it's possible to cache only GETs to a specific endpoint, since we sometimes use the same url for both GETs and POSTs

Reply:
++

This would allow us to get rid of [this check](https://github.com/Yelp/casper/blob/f6a1dae8bfc7c7f023ea71e40ceff766ae3fd41f/lua/spectre_common.lua#L108) entirely and defer the cacheable/non-cacheable decision until later.

Potentially it's bad for performance if a service's traffic is mostly made of uncacheable methods. But now that we're allowing HEAD, GET, and POST...this early check doesn't buys us much anyway.

Reply:
@drolando I added the optional 'request_method' setting to the config. 

@ArnaudBrousseau  I was able to combine the check request method check with the pattern check, so uncacheable methods will be short-circuited in `determine_if_cacheable` method as before.